src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -33,75 +36,938 @@ const ActorInit En_In_InitVars = {
+            if (gSaveContext.infTable[0x9] & 0x400) {
+                return 0x2031;
+            } else {
+                return 0x2030;
+            }
+    }
+}
+
+u16 func_80A79168(GlobalContext* globalCtx, Actor* thisx) {
+    u16 temp_v0 = Text_GetFaceReaction(globalCtx, 25);
+
+    if (temp_v0 != 0) {
+        return temp_v0;
+    }
+    if (LINK_IS_CHILD) {
+        return func_80A78FB0(globalCtx);
+    } else {
+        return func_80A79010(globalCtx);
+    }
+}
+
+s16 func_80A791CC(GlobalContext* globalCtx, Actor* thisx) {
+    s32 temp_var = 0;
+    switch (thisx->textId) {
+        case 0x2045:
+            gSaveContext.infTable[9] |= 0x80;
+            break;
+        case 0x203E:
+            temp_var = 2;
+            break;
+        case 0x203F:
+            gSaveContext.eventChkInf[1] |= 2;
+            gSaveContext.infTable[9] |= 0x10;
+            break;
+    }
+    return temp_var;
+}
+
+s16 func_80A7924C(GlobalContext* globalCtx, Actor* thisx) {
+    EnIn* this = THIS;
+    s32 sp18 = 1;
+
+    switch (this->actor.textId) {
+        case 0x2030:
+        case 0x2031:
+            if (globalCtx->msgCtx.choiceIndex == 1) {
+                this->actor.textId = 0x2032;
+            } else if (gSaveContext.rupees < 10) {
+                this->actor.textId = 0x2033;
+            } else {
+                this->actor.textId = 0x2034;
+            }
+            func_8010B720(globalCtx, this->actor.textId);
+            gSaveContext.infTable[9] |= 0x400;
+            break;
+        case 0x2034:
+            if (globalCtx->msgCtx.choiceIndex == 1) {
+                Rupees_ChangeBy(-10);
+                this->actor.textId = 0x205C;
+            } else {
+                this->actor.textId = 0x2035;
+            }
+            func_8010B720(globalCtx, this->actor.textId);
+            break;
+        case 0x2036:
+        case 0x2037:
+            if (globalCtx->msgCtx.choiceIndex == 1) {
+                sp18 = 2;
+            } else {
+                this->actor.textId = 0x201F;
+                func_8010B720(globalCtx, this->actor.textId);
+            }
+            break;
+        case 0x2038:
+            if (globalCtx->msgCtx.choiceIndex == 0 && gSaveContext.rupees >= 50) {
+                sp18 = 2;
+            } else {
+                this->actor.textId = 0x2039;
+                func_8010B720(globalCtx, this->actor.textId);
+                gSaveContext.infTable[10] |= 4;
+            }
+            break;
+        case 0x205B:
+            if (globalCtx->msgCtx.choiceIndex == 0 && gSaveContext.rupees >= 50) {
+                sp18 = 2;
+            } else {
+                func_8010B720(globalCtx, this->actor.textId = 0x2039);
+                gSaveContext.eventInf[0] &= ~0xF;
+                gSaveContext.eventInf[0] &= ~0x20;
+                gSaveContext.eventInf[0] &= ~0x40;
+                this->actionFunc = func_80A7A4C8;
+            }
+            break;
+    }
+    if (!gSaveContext.rupees) {}
```

Comment:
void0 might help get rid of this?

Reply:
Didn't seem to help in this case unfortunately.